Userland/Libraries/LibWeb/MimeSniff/Resource.cpp

```diff
@@ -7,6 +7,256 @@
+
+            // The case-insensitive string "<TITLE" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x54\x49\x54\x4C\x45\x00"sv.bytes(),
+                "\xFF\xDF\xDF\xDF\xDF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<B" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x42\x00"sv.bytes(), "\xFF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<BODY" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x42\x4F\x44\x59\x00"sv.bytes(), "\xFF\xDF\xDF\xDF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<BR" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x42\x52\x00"sv.bytes(), "\xFF\xDF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<P" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x50\x00"sv.bytes(), "\xFF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The string "<!--" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x21\x2D\x2D\x00"sv.bytes(), "\xFF\xFF\xFF\xFF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The string "<?xml".
+            BytePatternTableRow { "\x3C\x3F\x78\x6D\x6C"sv.bytes(), "\xFF\xFF\xFF\xFF\xFF"sv.bytes(), ignored_whitespace_bytes, "text/xml"sv },
+
+            // The string "%PDF-", the PDF signature.
+            BytePatternTableRow { "\x25\x50\x44\x46\x2D"sv.bytes(), "\xFF\xFF\xFF\xFF\xFF"sv.bytes(), no_ignored_bytes, "application/pdf"sv },
+        };
+
+        for (auto const& row : pattern_table) {
+            // 1. Let patternMatched be the result of the pattern matching algorithm given resource’s resource header,
+            //    the value in the first column of row, the value in the second column of row, and the value in the
+            //    third column of row.
+            auto pattern_matched = pattern_matching_algorithm(resource.resource_header(), row.byte_pattern(), row.pattern_mask(), row.ignored_leading_bytes(), row.is_tag_terminated());
+
+            // 2. If patternMatched is true, return the value in the fourth column of row.
+            if (pattern_matched) {
+                if (auto maybe_type = TRY(MimeType::parse(row.mime_type())); maybe_type.has_value())
+                    return maybe_type.release_value();
+            }
+        }
+    }
+
+    // 2. Execute the following steps for each row row in the following table:
+    static auto text_plain_mime_type = "text/plain"sv;
+    static Array<BytePatternTableRow, 4> pattern_table {
```

Comment:
these should be constexpr

Reply:
Many thanks for reviewing this @ADKaster! Currently the only problem making the tables constexpr is that "..."sv.bytes() returns a non-constexpr result so I'll look into how I can pass these bytes to the `BytePatternTableRow` some other way

Reply:
Recent change resolves all of the requested changes except this one. Still unsure how to pass constexpr values to `BytePatternTableRow`. I could just add static variables for each byte pattern/pattern mask pair and pass them but that would make the code quite icky. Will continue to look for other solutions

Reply:
Pushed new changes; my current approach is just to have `BytePatternRow` use `StringView` for both `byte_pattern` and `pattern_mask` so that we may use string view literals as constexpr values. I'll keep this open in case a better approach is found